sotodlib/tod_ops/jumpfind.py

```diff
@@ -0,0 +1,406 @@
+import numpy as np
+import scipy.signal as sig
+import scipy.ndimage as simg
+from so3g.proj import RangesMatrix
+from skimage.restoration import denoise_tv_chambolle
+
+
+def _jumpfind(x, min_chunk, min_size, win_size, max_depth=-1, depth=0, **kwargs):
+    """
+    Recursive edge detection based jumpfinder.
+
+    Note that the jumpfinder is very sensitive to changes in parameters
+    and the parameters are not independant of each other,
+    so it may take some playing around to get it to work properly.
+
+    Arguments:
+
+        x: Data to jumpfind on, expects 1D
+           Note that x will by modified in place with jumps removed
+           pass x.copy() to preserve x (say to pass to a better jump fixer)
+
+        min_chunk: The smallest chunk of data to look for jumps in
+
+        min_size: The smalled jump size counted as a jump
+
+        win_size: Number of samples to average over when checking jump size
+
+        max_depth: The maximum recursion depth, set negetive to not use
+
+        depth: The current recursion depth
+
+        **kwargs: Arguments to pass to scipy.signal.find_peaks
+
+    Returns:
+
+        jumps: The indices of jumps in x
+               There is some uncertainty on order of 1 sample
+               Jumps within min_chunk of each other may not be distinguished
+    """
+    if len(x) < min_chunk:
+        return np.array([], dtype=int)
+
+    # If std is basically 0 no need to check for jumps
+    if np.isclose(x.std(), 0.0):
+        return np.array([], dtype=int)
+
+    # Mean subtract the data
+    _x = x - x.mean()
+
+    # Scale data to have std of order 1
+    if _x.std() > 10:
+        _x = _x / (10 ** (int(np.log10(_x.std()))))
+
+    # Take cumulative sum
+    x_step = np.cumsum(_x)
```

Comment:
Add to comment that this is equivalent to convolution with a step function, as one might do in a matched filter (or something to that effect).  I think you should also move the mean removal closer, since it's essential to this method's success.

(Thanks for the off-line explanation of this!)